subsys/bluetooth/controller/hal/nrf5/radio/radio.c

```diff
@@ -890,27 +916,52 @@ void radio_tmr_end_capture(void)
 	HAL_RADIO_END_TIME_CAPTURE_PPI_REGISTER_TASK =
 		HAL_RADIO_END_TIME_CAPTURE_PPI_TASK;
 	NRF_PPI->CHENSET = HAL_RADIO_END_TIME_CAPTURE_PPI_ENABLE;
+
 #if defined(CONFIG_BOARD_NRFXX_NWTSIM)
 	NRF_PPI_regw_sideeffects();
 #endif
+
+#if  defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+	HAL_EVENT_TIMER_CLEAR_PPI_REGISTER_EVT = HAL_EVENT_TIMER_CLEAR_PPI_EVT;
+	HAL_EVENT_TIMER_CLEAR_PPI_REGISTER_TASK =
+		HAL_EVENT_TIMER_CLEAR_PPI_TASK;
+	NRF_PPI->CHENSET = HAL_EVENT_TIMER_CLEAR_PPI_ENABLE;
+
+#if defined(CONFIG_BOARD_NRFXX_NWTSIM)
+	NRF_PPI_regw_sideeffects();
+#endif
+#endif /* defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) */
 }
 
 u32_t radio_tmr_end_get(void)
 {
+#if defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+	return last_pdu_end_us;
+#else /* defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) */
 	return EVENT_TIMER->CC[2];
+#endif /* defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) */
 }
 
+static u32_t tmr_sample_val;
 void radio_tmr_sample(void)
 {
-	EVENT_TIMER->TASKS_CAPTURE[3] = 1;
+	u32_t cc;
```

Comment:
Missing conditional compilation.

Keep the old implementation as is when the new feature is not used, it has lot lesser influence to measurement.